tests/phpunit/tests/filesystem/wpMkdirP.php

```diff
@@ -0,0 +1,171 @@
+		mkdir( self::$test_directory );
+	}
+
+	/**
+	 * Deletes the contents of the test directory after each test runs.
+	 */
+	public function tear_down() {
+		foreach ( $this->files_in_dir( self::$test_directory ) as $file ) {
+			$this->unlink( $file );
+		}
+
+		$matched_dirs = $this->scandir( self::$test_directory );
+		foreach ( array_reverse( $matched_dirs ) as $dir ) {
+			rmdir( $dir );
+		}
+
+		parent::tear_down();
+	}
+
+	/**
+	 * Deletes the test directory after all tests have run.
+	 */
+	public static function tear_down_after_class() {
+		rmdir( self::$test_directory );
+
+		parent::tear_down_after_class();
+	}
+
+	/**
+	 * Tests that `wp_mkdir_p()` fires an action.
+	 *
+	 * @ticket 44083
+	 *
+	 * @dataProvider data_actions
+	 *
+	 * @param string $hook_name The name of the action hook.
+	 */
+	public function test_wp_mkdir_p_should_fire_action( $hook_name ) {
+		$action = new MockAction();
+		add_action( $hook_name, array( $action, 'action' ) );
+
+		wp_mkdir_p( self::$test_directory . $hook_name );
+
+		$this->assertSame( 1, $action->get_call_count() );
+	}
+
+	/**
+	 * Tests that `wp_mkdir_p()` does not fire an action when a file exists of the same name.
+	 *
+	 * @ticket 44083
+	 *
+	 * @dataProvider data_actions
+	 *
+	 * @param string $hook_name The name of the action hook.
+	 */
+	public function test_wp_mkdir_p_should_not_fire_action_when_a_file_exists_of_the_same_name( $hook_name ) {
+		$action = new MockAction();
+		add_action( $hook_name, array( $action, 'action' ) );
+
+		$target = self::$test_directory . $hook_name;
+
+		// Force a failure.
+		$this->touch( $target );
+
+		wp_mkdir_p( $target );
+
+		$this->assertSame( 0, $action->get_call_count() );
+	}
+
+	/**
+	 * Tests that `wp_mkdir_p()` does not fire an action when the target directory contains '../'.
+	 *
+	 * @ticket 44083
+	 *
+	 * @dataProvider data_actions
+	 *
+	 * @param string $hook_name The name of the action hook.
+	 */
+	public function test_wp_mkdir_p_should_not_fire_action_when_the_target_contains_path_traversal( $hook_name ) {
+		$action = new MockAction();
+		add_action( $hook_name, array( $action, 'action' ) );
+
+		// Force a failure by not using `realpath()`.
+		$target = DIR_TESTDATA . "/$hook_name/";
+
+		wp_mkdir_p( $target );
+
+		$this->assertSame( 0, $action->get_call_count() );
+	}
+
+	/**
+	 * Tests that `wp_mkdir_p()` does not fire an action when the target directory contains '..DIRECTORY_SEPARATOR'.
+	 *
+	 * @ticket 44083
+	 *
+	 * @dataProvider data_actions
+	 *
+	 * @param string $hook_name The name of the action hook.
+	 */
+	public function test_wp_mkdir_p_should_not_fire_action_when_the_target_contains_path_traversal_with_directory_separator( $hook_name ) {
+		$action = new MockAction();
+		add_action( $hook_name, array( $action, 'action' ) );
+
+		// Force a failure.
+		$target = str_replace( '../', '..' . DIRECTORY_SEPARATOR, DIR_TESTDATA ) . "/$hook_name";
```

Comment:
```suggestion
		$target = str_replace( '../', '..' . DIRECTORY_SEPARATOR, DIR_TESTDATA ) . "/test_wp_mkdir_p/$hook_name";
```
Same as before: Adding parent.